ush/soca/run_jjobs.py

```diff
@@ -0,0 +1,155 @@
+homegfs = exp_config['gw environement']['experiment identifier']['HOMEgfs']
+stmp = exp_config['gw environement']['working directories']['STMP']
+rotdirs = exp_config['gw environement']['working directories']['ROTDIR']
+expdirs = exp_config['gw environement']['working directories']['EXPDIRS']
+
+# Generate the setup_expt.sh script
+with open("setup_expt.sh", "w") as bash_script:
+    bash_script.write("#!/usr/bin/env bash\n")
+    export_env_vars_script(exp_config['gw environement'], bash_script, pslot)
+
+    # Make a copy of the configs
+    origconfig = "${HOMEgfs}/parm/config"
+    bash_script.write("\n")
+    bash_script.write("# Make a copy of config\n")
+    bash_script.write(f"cp -r {origconfig} .\n")
+
+    # Dump the configs in a separate yaml file
+    with open("overwrite_defaults.yaml", "w") as f:
+        yaml.safe_dump(exp_config['setup_expt config'], f)
+
+    # Setup the experiment
+    bash_script.write("\n")
+    bash_script.write("# Setup the experiment\n")
+
+    setupexpt = "${HOMEgfs}/workflow/setup_expt.py cycled "
+    # Most of the args keys are not used to run the jjobs but are needed to run setup_expt.py
+    args = {
+        "idate": "${PDY}${cyc}",
+        "edate": "${PDY}${cyc}",
+        "app": "ATM",
+        "start": "warm",
+        "gfs_cyc": "0",
+        "resdet": exp_config['setup_expt config']['resdet'],
+        "resens": "24",
+        "nens": "0",
+        "pslot": "${PSLOT}",
+        "configdir": "${PWD}/config",
+        "comrot": "${ROTDIR}",
+        "expdir": "${EXPDIRS}",
+        "yaml": "overwrite_defaults.yaml"}
+
+    # Write the arguments of setup_expt.py
+    for key, value in args.items():
+        setupexpt += f" --{key} {value}"
+    bash_script.write(f"{setupexpt}\n")
+    bash_script.close()
+
+# Execute prep.sh
+execute_script('setup_expt.sh')
+
+# get the machine value ... Do we still need it?
+machine = exp_config['machine']
+
+configbase = os.path.join(exp_config['gw environement']['working directories']['EXPDIRS'],
+                          exp_config['gw environement']['experiment identifier']['PSLOT'],
+                          'config.base')
+
+# Append SLURM directive so the script can be "sbatch'ed"
+machines = {"container", "hera", "orion"}
+if machine in machines:
+    print(f'machine is {machine}')
+else:
+    print(f"Probably does not work for {machine} yet")
+
+# swap a few variables in config.base
+var2replace = {'HOMEgfs': homegfs, 'STMP': stmp, 'ROTDIRS': rotdirs, 'EXPDIRS': expdirs}
+with open(configbase, 'r') as f:
+    newconfigbase = f.read()
+for key, value in var2replace.items():
+    newconfigbase = newconfigbase.replace(f'{key}=', f'{key}={value} #')
+with open(configbase, 'w') as f:
+    f.write(newconfigbase)
+
+# Generate the script that runs the jjobs
+with open("run_jjobs.sh", "w") as bash_script:
+    bash_script.write("#!/usr/bin/env bash\n")
+    if machine != "container":
+        slurm_header(exp_config['job options'], bash_script)
```

Comment:
This is OK since at present we can only run JEDI on Hera and Orion.  Both systems use slurm.   We will need to generalize this when we can run JEDI on WCOSS2.  WCOSS2 uses PBS.